ledger/src/blockstore.rs

```diff
@@ -3964,6 +3964,13 @@ impl Blockstore {
         Ok(duplicate_slots_iterator.map(|(slot, _)| slot))
     }
 
+    pub fn has_existing_shreds_for_slot(&self, slot: Slot) -> bool {
+        match self.meta(slot).unwrap() {
```

Comment:
why unwrap here?

Reply:
I think is just replicating the current unwrap behavior. Looks like this fails if we:

1. hit a rocksdb error
2. fail deserialization

Neither of these should happen in "normal" operation. Expect would be better.